cluster-api/cloud/google/metadata.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"k8s.io/kube-deploy/cluster-api/cloud/google/machinesetup"
+	clusterv1 "k8s.io/kube-deploy/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+type metadataParams struct {
+	Token        string
+	Cluster      *clusterv1.Cluster
+	Machine      *clusterv1.Machine
+	DockerImages []string
+	Project      string
+	Metadata     *machinesetup.Metadata
+
+	// These fields are set when executing the template if they are necessary.
+	PodCIDR        string
+	ServiceCIDR    string
+	MasterEndpoint string
+}
+
+func nodeMetadata(params metadataParams) (map[string]string, error) {
+	params.PodCIDR = getSubnet(params.Cluster.Spec.ClusterNetwork.Pods)
+	params.ServiceCIDR = getSubnet(params.Cluster.Spec.ClusterNetwork.Services)
+	params.MasterEndpoint = getEndpoint(&params.Cluster.Status.APIEndpoints[0])
+
```

Comment:
(nit)
Perhaps the getEndpoint method should take a clusterv1.APIEndpoint instead of a *clusterv1.APIEndpoint?